File: ERPServer/Shared/Shared/Entities/Purchases/PurchasesBillHeaderDTO.cs
using Data.Entities.Shared;
using Newtonsoft.Json;
using Shared.Entities.Shared;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Shared.Entities.Purchases
{
    public class PurchasesBillHeaderDTO : AuditEntityDTO
    {
        public bool IsActive { get; set; }
        public string Number { get; set; }
        public string Date { get; set; }
        public decimal Total { get; set; }
        public decimal Discount { get; set; }
        public decimal OtherExpenses { get; set; }
        public decimal TotalAfterDiscount { get; set; }
        public decimal TotalDiscount { get; set; }
        public decimal VatAmount { get; set; }
        public decimal TotalAfterVAT { get; set; }
        public decimal TaxPercentage { get; set; }
        public decimal TaxAmount { get; set; }
        public decimal TotalAmount { get; set; }
        public decimal Paid { get; set; }
        public decimal Remaining { get; set; }
        public bool IsTax { get; set; }
        public bool IsTemp { get; set; }
        public bool IsCancel { get; set; }
        public bool IsReturned { get; set; }
        public bool IsNewReturned { get; set; }
        public string Notes { get; set; }
        public long ClientVendorId { get; set; }
        public long? RepresentiveId { get; set; }
        public string ClientVendorName { get; set; }
        public long? CompanyId { get; set; }
        public long? TreasuryId { get; set; }
        public long? ParentId { get; set; }

        public virtual List<PurchasesBillDetailDTO> PurchasesBillDetailList { get; set; }

    }
}

File: erp-app/src/app/modules/purchases/models/purchases-bill-header.dto.ts
import { AuditDTO } from "src/app/shared/models/audit-dto.model";
import { PurchasesBillDetailsDTO } from "./purchases-bill-details.dto";
import { PaymentMethodEnum } from "src/app/shared/enums/payment-method.enum";

export class PurchasesBillHeaderDTO extends AuditDTO {
    id: number | null;
    number: string;
    isActive: boolean;
    date: string | undefined;
    clientVendorId: number | null;
    clientVendorName: string;
    notes: string;
    companyId: number;
    representiveId: number | null;
    discount: number = 0 | 0;
    otherExpenses: number = 0 | 0;
    total: number = 0 | 0;
    totalAfterDiscount: number = 0 | 0;
    vatAmount: number = 0 | 0;
    totalAfterVAT: number = 0 | 0;
    taxPercentage: number = 0 | 0;
    taxAmount: number = 0 | 0;
    totalAmount: number = 0 | 0;
    paid: number;
    remaining: number = 0 | 0;
    isTax: boolean = false;
    purchasesBillDetailList: Array<PurchasesBillDetailsDTO> = Array<PurchasesBillDetailsDTO>();
    removedPurchasesBillDetailList: Array<PurchasesBillDetailsDTO> = Array<PurchasesBillDetailsDTO>();
    isTemp: boolean = false;
    isReturned: boolean = false;
    isNewReturned: boolean = false;
    paymentMethodId: PaymentMethodEnum;
    refNo: string;
}